cmd/loki/main.go

```diff
@@ -28,7 +30,12 @@ func main() {
 	)
 	flag.StringVar(&configFile, "config.file", "", "Configuration file to load.")
 	flagext.RegisterFlags(&cfg)
-	flag.Parse()
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.Parse()
+	if err := viper.BindPFlags(pflag.CommandLine); err != nil {
```

Comment:
After that you should load the config file using viper.

See https://gist.github.com/chazcheadle/45bf85b793dea2b71bd05ebaa3c28644

Reply:
same goes for promtail.

This way we can solve the precedence issue with flag and the config file.

ps: can we make it strict ? if a config is found with no match in the struct it should fail at least that the current behaviour.

Reply:
met issue:
```
&{{{  80  9095 true false 30s 30s 30s 2m0s [] [] [] [] 4194304 4194304 100 {info info 0x85ec40} {0xc0004bca20} } } { 1s 102400 {}
 {100ms 5s 5} [lb1=v1,lb2=v2] 10s} [] {10s /var/log/positions.yaml} [] {10s}}
level=error ts=2019-06-14T01:31:29.781042771Z caller=main.go:62 msg="error creating promtail" error="at least one client config should be provided"
```
Those info printed by `viper.Debug`.

Reply:
>  if a config is found with no match in the struct it should fail at least that the current behaviour.
Sure, viper auto support it.